tests/async/test_queryselector.py

```diff
@@ -179,3 +179,398 @@ async def test_selectors_register_should_handle_errors(selectors, page: Page, ut
     with pytest.raises(Error) as exc:
         await selectors.register("css", dummy_selector_engine_script)
     assert exc.value.message == '"css" is a predefined selector engine'
+
+
+async def test_should_work_with_layout_selectors(page: Page) -> None:
```

Comment:
Ported from https://github.com/microsoft/playwright/blob/ba0cfaeb2df509e4be93bfc950a4f43a913a2955/tests/page/selectors-misc.spec.ts#L165.

We don't really need _all_ these in the python repo but all are easy enough to port.

Reply:
I did the same for .NET, sounds good!